src/http/Response.js

```diff
@@ -71,4 +71,19 @@ export default class Response extends Message {
     prefix(messageWriter) {
         return messageWriter.responsePrefix(this);
     }
+
+    // returns the 'ID' field value of the response copied to from the request (if any)
+    extractMatchingId(request) {
```

Comment:
Do the callers need a _matching_ request ID?

* If yes, this code is missing a check that the request ID value in the response matches the given request ID value. I would rename the method to matchesRequestId() in this case, and have it return a boolean value.
* Otherwise, we should rename this method to requestId(), I think.